analyses/molecular-subtyping-LGAT/03-subset-cnv-files-LGAT.Rmd

```diff
@@ -0,0 +1,329 @@
+
+overlaps_manta_fgfr_df
+```
+4 biospecimens have tandem duplication that overlap tyrosine kinase domain in FGFR1
+
+
+### Distribution of copy number
+CNV that are within FGFR1 gene region only and overlap "Protein tyrosine kinase" domain in FGFR1 in biospecimens
+
+```{r}
+
+# gene region overlap
+fgfr_gene <- GenomicRanges::GRanges(
+  seqnames = paste0("chr", domain_gr$chromosome_name),
+  ranges = IRanges::IRanges(start = domain_gr$gene_start, end = domain_gr$gene_end)
+)
+
+# subset cnv+domain grange to be within FGFR1 gene region
+cnv_fgfr_gr <- IRanges::subsetByOverlaps(cnv_gr, fgfr_gene, type = "within", maxgap = )
+
+# overlap consensus seg with exon
+overlaps_fgfr_cnv <- IRanges::subsetByOverlaps(cnv_fgfr_gr, domain_gr)
+
+# filter for fgfr only overlap
+overlaps_fgfr_cnv <- overlaps_fgfr_cnv[which(
+  # filter for duplication
+  overlaps_fgfr_cnv$copy_number > 2
+), ]
+
+# dataframe
+overlaps_cnv_df <- data.frame(
+  biospecimen_id = overlaps_fgfr_cnv$ID,
+  copy_number = overlaps_fgfr_cnv$copy_number,
+  stringsAsFactors = FALSE
+) %>%
+  dplyr::distinct() %>%
+  # keep copy_number > 2
+  dplyr::mutate(FGFR_DUP = if_else(copy_number > 2, "Yes", "No"))
+
+overlaps_cnv_df
+```
+5 samples have copy_number changes > 2 that overlap with tyrosine kinase domain
+
+Are there overlap in SV and CNV calls?
+
+```{r}
+intersect(overlaps_manta_fgfr_df$biospecimen_id, overlaps_cnv_df$biospecimen_id)
+```
+Only BS_MW6YMRBJ shows tandem duplication in manta and duplication in consensus seg file that overlaps tyrosine kinase domain in FGFR1
+
+### Checking which exons overlap the above SV and CNV
+
+```{r}
+# Define the annotations for the hg38 genome
+txdb <- GenomicFeatures::makeTxDbFromGFF(
+  file = file.path(
+    root_dir, "data",
+    "gencode.v27.primary_assembly.annotation.gtf.gz"
+  ),
+  format = "gtf"
+)
+# get exons and exon_id for FGFR1 canonical transcript
+fgfr1_exons <- GenomicFeatures::exons(txdb,
+  columns = list("exon_rank"),
+  # genocode canonical transcript https://genome.ucsc.edu/cgi-bin/hgGene?db=hg38&hgg_gene=FGFR1
+  filter = list(tx_name = "ENST00000447712.6")
+)
+```
+
+For manta SV calls that are within FGFR1 gene region
+
+```{r}
+# overlap filtered manta with exons
+overlaps_manta_fgfr_exon <- IRanges::mergeByOverlaps(overlaps_manta_fgfr, fgfr1_exons)
+
+# overlap filtered cnv with exons
+overlaps_cnv_fgfr_exon <- IRanges::mergeByOverlaps(overlaps_fgfr_cnv, fgfr1_exons)
+
+# plot exon rank for manta and cnv
+ggplot(
+  rbind(
+    data.frame(
+      "exon_number" = unlist(overlaps_manta_fgfr_exon$exon_rank),
+      "type" = "tandem_dup"
+    ),
+    data.frame(
+      "exon_number" = unlist(overlaps_cnv_fgfr_exon$exon_rank),
+      "type" = "cnv_dup"
+    )
+  ),
+  aes(x = exon_number, color = type)
+) +
+  geom_density()
+```
+Majority SV and CNV overlap exons downstream of exon 5
+
+any SV overlap which overlaps exons 10 through 18 
+```{r}
+overlaps_manta_fgfr_exon[all(overlaps_manta_fgfr_exon$exon_rank >= 10 &
```

Comment:
One thing I'm not clear on here: is there a reason we want to exclude if more exons than 10 -18 are duplicated? aka if a CNV or SV has 8- 20 duplicated we would be excluding that here, is there a reason we only want 10 -18 but not more or less than that?

Reply:
This and the 1 below were just checks to see if the CNV/SV  regions captured from the steps above overlap 10-18 exon region as we see in previous references mentioned in #790. The CNV/SVs do actually cover regions but 2 bs_ids overlap 9 exon as well. 

Instead of these checks I've just used the ggplot to specify the same information.

Reply:
This one also resolved @cansavvy ?

Reply:
Sounds good.